File: Java/Hussain Sir Java/Fundamentals/Runtime/Reflection/Interval.java
class Interval{

	private int min;
	private int sec;

	public Interval(int m, int s){
		min = m + s / 60;
		sec = s % 60;
	}

	public int minutes(){
		return min;
	}

	public int seconds(){
		return sec;
	}

	public int time(){
		return 60 * min + sec;
	}

	public String toString(){
		return String.format("%d:%02d", min, sec);
	}

	public int hashCode(){
		return min + sec;
	}

	public boolean equals(Object other){
		if(other instanceof Interval){
			Interval that = (Interval)other;
			return (this.min == that.min) && (this.sec == that.sec); 
		}
		return false;
	}

}

File: Fundamentals/Generics/Collections/interval.cs
using System;

partial class Interval : IComparable<Interval>
{
	public int Minutes {get;}

	public int Seconds {get;}

	partial void OnCreate();

	public Interval(int minutes, int seconds)
	{
		Minutes = minutes + seconds / 60;
		Seconds = seconds % 60;
		OnCreate();
	}

	public int GetTime()
	{
		return 60 * Minutes + Seconds;
	}

	public override string ToString()
	{
		return $"{Minutes}:{Seconds:00}";
	}

	public override int GetHashCode() => Minutes + Seconds;
	
	public override bool Equals(object other)
	{
		if(other is Interval that)
		{
			return this.Minutes == that.Minutes && this.Seconds == that.Seconds;
		}

		return false;
	}

	public int CompareTo(Interval that)
	{
		return this.GetTime() - that.GetTime();
	}
}

